File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] indices = new int[2];
        for (int i = 0; i < nums.length; i++) {
            int num1 = nums[i];
            for (int j = i+1; j < nums.length; j++) {
                int num2 = nums[j];
                if (num1 + num2 == target) {
                    indices[0] = i;
                    indices[1] = j;
                }
            }
        }
        return indices;

    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []

        def dfs(index, current_subset, total):
            if total == target:
                res.append(current_subset.copy())
                return

            if index >= len(nums) or total > target:
                return

            current_subset.append(nums[index])
            dfs(index, current_subset, (total + nums[index]))
            current_subset.pop()
            dfs(index + 1, current_subset, total)
        
        dfs(0, [], 0)
        return res